packages/@coorpacademy-player-store/src/actions/api/progressions.js

```diff
@@ -23,29 +26,49 @@ import {
   getCurrentSlide,
   getPreviousSlide
 } from '../../utils/state-extract';
-import type {
-  Action,
-  Dispatch,
-  DispatchedAction,
-  GetState,
-  ThunkAction
-} from '../../definitions/redux';
-
-export const PROGRESSION_FETCH_REQUEST: string = '@@progression/FETCH_REQUEST';
-export const PROGRESSION_FETCH_SUCCESS: string = '@@progression/FETCH_SUCCESS';
-export const PROGRESSION_FETCH_FAILURE: string = '@@progression/FETCH_FAILURE';
+import type {Dispatch, DispatchedAction, GetState, ThunkAction} from '../../definitions/redux';
 
-export type ProgressionAction = {
-  ...Action,
-  payload: Progression
+export type Action = {
+  type: '@@progression/FETCH_SUCCESS' | '@@progression/CREATE_SUCCESS',
+  payload: Progression,
+  meta?: {
+    id?: ProgressionId,
+    progressionId?: ProgressionId
+  }
 };
 
-export type FetchSuccessAction = {
-  ...Action,
-  payload: Progression,
-  type: '@@progression/FETCH_SUCCESS'
+export const PROGRESSION_CREATE_REQUEST: string = '@@progression/CREATE_REQUEST';
+export const PROGRESSION_CREATE_SUCCESS: string = '@@progression/CREATE_SUCCESS';
+export const PROGRESSION_CREATE_FAILURE: string = '@@progression/CREATE_FAILURE';
+
+export const createProgression = (
+  engine: Engine,
+  content: Content,
+  engineOptions: EngineOptions
+): ThunkAction => async (
+  dispatch: Function,
+  getState: GetState,
+  {services}: {services: Services}
+): // $FlowFixMe circular declaration issue with gen-flow-files : type ThunkAction = (Dispatch, GetState, Options) => DispatchedAction
+DispatchedAction => {
```

Comment:
Actions should definitely return `Action` instead of dispatched ones, this is the role of middleware to do that. This is why you have circular issues. Every store property (actions+reducer) should live alone, and don't need to share any type. But at this time, this is ok if you understand the arguments of why it's not a good idea to do that in the future